pints/toy/_cone.py

```diff
@@ -0,0 +1,72 @@
+#
+# Unimodal Normal/Gaussian toy log pdf.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+import scipy
+
+
+class ConeLogPDF(pints.LogPDF):
+    """
+    Toy distribution based on a d-dimensional distribution of the form,
+
+    .. math::
+        f(x) \propto e^{-|x|^\beta}
+
+    where x is a d-dimensional real, and |x| is the Euclidean norm. The mean
+    and variance that are returned relate to expectations on |x| not the
+    multidimensional x.
+
+    Arguments:
+
+    ``dimensions``
+        The dimensionality of the cone.
+    ``beta``
+        The power to which |x| is raised in the exponential term, which must be
+        positive.
+
+    *Extends:* :class:`pints.LogPDF`.
+    """
+    def __init__(self, dimensions=2, beta=1):
+        if dimensions < 1:
+            raise ValueError('Dimensions must not be less than 1.')
+        if not isinstance(dimensions, int):
```

Comment:
Easier to just write `dimensions = int(dimensions)` on the first line and just let it fail if it can't do the conversion! We don't want to do type checking really